### PR TITLE
Update BTApplePayCardNonce initializer

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes
* Remove `type` and `nonce` params on `BTApplePayCardNone` initializer

### Checklist

- [X] Added a changelog entry

### Authors
@sestevens @scannillo

### CODE REVIEW

File: Sources/BraintreeApplePay/Public/BraintreeApplePay/BTApplePayCardNonce.h [hunk 1]
```diff
@@ -16,10 +16,15 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly, strong) BTBinData *binData;
 
+/**
+ The last two digits of the card associated with this nonce, if available.
```
Comment (hollabaq86) [inline Sources/BraintreeApplePay/Public/BraintreeApplePay/BTApplePayCardNonce.h:20]:
```suggestion
 The last two digits of the Device Primary Account Number (DPAN) associated with this nonce, if available.
```

File: UnitTests/BraintreeApplePayTests/BTApplePayCardNonce_Tests.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import XCTest
+
+class BTApplePayCardNonce_Tests: XCTestCase {
+
+    func testInitWithNonce_populatesAllProperties() {
```
Comment (sshropshire) [inline UnitTests/BraintreeApplePayTests/BTApplePayCardNonce_Tests.swift:5]:
```suggestion
    func testInitJson_populatesAllProperties() {
```

**Review by hollabaq86 — COMMENTED** [review]
I want to discuss the choice of the name `dpanLastTwo`. Are we including `dpan` in the name to make sure folks understand it's slightly different than [`lastTwo` in `BTCardNonce`](https://github.com/braintree/braintree_ios/blob/7eb97ae53215d05425fce165d659753fa037f296/Sources/BraintreeCard/Public/BraintreeCard/BTCardNonce.h#L40)?

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreeApplePay/Public/BraintreeApplePay/BTApplePayCardNonce.h (hunk 1 above)
Comment (scannillo) [reply Sources/BraintreeApplePay/Public/BraintreeApplePay/BTApplePayCardNonce.h]: We removed this property entirely. We originally thought DPAN was the same as PAN.

**Comment by scannillo** [comment]
We updated this PR to remove the `dpanLastTwo` property on `BTApplePayCardNonce`. This PR now just updates the initializer to accept JSON as the only param. Thanks for the feedback!

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**